lua/modes/utils.lua

```diff
@@ -47,7 +47,8 @@ end
 ---Set highlight
 ---@param name string
 ---@param color Color
-M.set_hl = function(name, color)
+---@param if_not_exist? boolean
+M.set_hl = function(name, color, if_not_exist)
```

Comment:
More nitpicking. Keep your naming if you prefer it 😌 

```suggestion
---@param if_not_exists? boolean
M.set_hl = function(name, color, if_not_exists)
```